core/src/raw/adapters/kv/api.rs

```diff
@@ -25,10 +26,69 @@ use crate::Capability;
 use crate::Scheme;
 use crate::*;
 
+/// ScanIter is the async iterator returned by `Adapter::scan`.
+pub trait Scan: Send + Sync + Unpin {
+    /// Fetch the next key in the current key prefix
+    ///
+    /// `None` means no further key will be returned
+    fn next(&mut self) -> impl Future<Output = Option<Result<String>>> + MaybeSend;
```

Comment:
Let's return `Result<Option<String>>` instead to make it easier to handling errors.

Reply:
Done.